main/src/com/google/refine/operations/column/ColumnAdditionByFetchingURLsOperation.java

```diff
@@ -134,8 +151,34 @@ public ColumnAdditionByFetchingURLsOperation(
         _delay = delay;
         _cacheResponses = cacheResponses;
         _httpHeadersJson = httpHeadersJson;
+
+        List<Header> headers = new ArrayList<Header>();
+        if (_httpHeadersJson != null) {
+            for (HttpHeader header : _httpHeadersJson) {
+                if (header.name != null && header.name != "" && header.value != null && header.value != "") {
+                    headers.add(new BasicHeader(header.name, header.value));
+                }
+            }
+        }
+        httpHeaders = headers.toArray(httpHeaders);
+
+        defaultRequestConfig = RequestConfig.custom()
+                .setConnectTimeout(30 * 1000)
+                .setConnectionRequestTimeout(30 * 1000)
+                .setSocketTimeout(10 * 1000).build();
+
+//        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+//        credsProvider.setCredentials(new AuthScope(host, 443),
+//                new UsernamePasswordCredentials(user, password));
```

Comment:
Why is this code commented out?

Reply:
It's not part of the current functionality, so not included in the refactoring. I left it as a placeholder for future feature enhancement, but I can remove it if you think that would be cleaner.

Reply:
Ah yes, I thought we supported this already, but we only expose an `Authorization` header in the UI. I generally try to avoid committing commented out code, but if you want to leave it as a demo for future improvements, it could be worth explaining that intention with a comment, perhaps?

Reply:
Comment added